go/vt/vttablet/tabletmanager/vreplication/vreplicator.go

```diff
@@ -179,7 +228,18 @@ func (vr *vreplicator) setMessage(message string) error {
 }
 
 func (vr *vreplicator) setState(state, message string) error {
-	return binlogplayer.SetVReplicationState(vr.dbClient, vr.id, state, message)
+	if message != "" {
+		vr.stats.History.Add(&binlogplayer.StatsHistoryRecord{
+			Time:    time.Now(),
+			Message: message,
+		})
+	}
+	vr.stats.State.Set(state)
```

Comment:
We have binlog player stats and vreplicator stats. How are the two intended to be used?

Reply:
They share the same structure. But they're updated differently depending on whether it's new vs legacy code.